src/server/services/wfs/qgswfsgetfeature.cpp

```diff
@@ -414,12 +414,29 @@ namespace QgsWfs
         geometryName = QLatin1String( "NONE" );
       }
       // outputCrs
-      QgsCoordinateReferenceSystem outputCrs = vlayer->crs();
+      // if the crs is defined in the parameters, use it
+      // otherwise fallback:
+      //  - geojson uses 'EPSG:4326' by default
+      //  - other formats use the default CRS (the layer's CRS)
+      const QString requestSrsName = request.serverParameters().value( QStringLiteral( "SRSNAME" ) );
+      QString outputSrsName;
       if ( !query.srsName.isEmpty() )
       {
-        outputCrs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( query.srsName );
+        outputSrsName = query.srsName;
+      }
+      else if ( !requestSrsName.isEmpty() )
+      {
+        outputSrsName = requestSrsName;
+      }
+      else
+      {
+        // fallback to a default value
+        // geojson uses 'EPSG:4326' by default
+        outputSrsName = ( aRequest.outputFormat == QgsWfsParameters::Format::GeoJSON ) ? QStringLiteral( "EPSG:4326" ) : vlayer->crs().authid();
       }
 
+      const QgsCoordinateReferenceSystem outputCrs = QgsCoordinateReferenceSystem::fromOgcWmsCrs( outputSrsName );
```

Comment:
Why not reusing QgsCoordinateReferenceSystem::createFromUserInput like it's done [here](https://github.com/qgis/QGIS/blob/master/src/server/services/wfs/qgswfsgetfeature.cpp#L827) instead ?

Reply:
We can't directly use `QgsCoordinateReferenceSystem::createFromUserInput` here because this is not a static method. Besides, `QgsCoordinateReferenceSystem::fromOgcWmsCrs` is also already used in `writeGetFeature`.

Reply:
We can use it this [way](https://github.com/qgis/QGIS/blob/1c903848de2/src/server/services/wfs/qgswfsgetfeature.cpp#L827) (in the same file). I'm a bit concerned that we don't use always the same method to load SRS but TBH, I don't know if it's normal, and I'm not sure which one would be the more appropriate.

Reply:
This is done now.